code/modules/mob/living/simple_animal/bot/bot.dm

```diff
@@ -899,6 +903,82 @@ Pass a positive integer as an argument to override a bot's default speed.
 			D.open()
 			frustration = 0
 
+/mob/living/simple_animal/bot/ui_data(mob/user)
+	var/list/data = list()
+	data["can_hack"] = (issilicon(user) || IsAdminGhost(user))
+	data["custom_controls"] = list()
+	data["emagged"] = emagged
+	data["locked"] = locked
+	data["pai"] = list()
+	data["settings"] = list()
+	if(!locked || issilicon(user) || IsAdminGhost(user))
+		data["pai"]["allow_pai"] = allow_pai
+		data["pai"]["card_inserted"] = paicard
+		data["settings"]["airplane_mode"] = !remote_disabled
+		data["settings"]["maintenance_lock"] = !open
+		data["settings"]["power"] = on
+		data["settings"]["booting"] = booting
+		data["settings"]["patrol_station"] = auto_patrol
+	return data
+
+// Actions received from TGUI
+/mob/living/simple_animal/bot/ui_act(action, params)
+	. = ..()
+	if(.)
+		return
+	if(!bot_core.allowed(usr) && !usr.has_unlimited_silicon_privilege)
```

Comment:
This is a much better check since it actually respects access requirements on borgs and AIs. Update the checks that are just checking for Issilicon and isadminghost to use this instead.